### PR TITLE
feat: Add nneg flag to InstCombine

### LINKED ISSUE #794: llvm: add `nneg` flag to zext
The following theorem currently fails:

```lean
theorem zext_nneg_sext_sgt_thm (e e_1 : IntW 8) :
  icmp IntPredicate.sgt (zext 32 e_1) (sext 32 e) ⊑ icmp IntPredicate.sgt e_1 e := by 
    simp_alive_undef
    simp_alive_ops
    simp_alive_case_bash
    simp_alive_split
    simp_alive_benchmark
    all_goals sorry
```

### PR DESCRIPTION
**Pull request by lfrenot**

### CODE REVIEW

**Comment by lfrenot** [comment]
This is not the full changes yet, but I'm having an issue I don't know how to fix

**Comment by tobiasgrosser** [comment]
@bollu @ineol, can you have a brief look?

File: SSA/Projects/InstCombine/Base.lean [hunk 1]
```diff
@@ -306,6 +308,9 @@ namespace Op
 @[match_pattern] abbrev icmp (c : IntPredicate)   : Nat → Op  := MOp.icmp c ∘ .concrete
 @[match_pattern] abbrev const (w : Nat) (val : ℤ) : Op        := MOp.const (.concrete w) val
 
+/- This operation is separate from the others because it takes in a flag: nneg. -/
+@[match_pattern] abbrev zext (w w': Nat) (flag: NonNegFlag := {nneg := false}) : Op := MOp.zext (.concrete w) (.concrete w') flag
```
Comment (bollu) [inline SSA/Projects/InstCombine/Base.lean]:
```suggestion
@[match_pattern] abbrev zext (w w': Nat) (flag : NonNegFlag := {nneg := false}) : Op := MOp.zext (.concrete w) (.concrete w') flag
```

**Review by bollu — APPROVED** [review]
LGTM! Thanks for the clean patch :)

**Pull request merged**